elasticsearch/src/main/java/org/hibernate/search/elasticsearch/client/impl/ArbitrarySort.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.elasticsearch.client.impl;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.hibernate.search.elasticsearch.impl.GsonHolder;
+
+import io.searchbox.core.search.sort.Sort;
+
+/**
+ * Implementation of a sort with arbitrary keys and values for Jest.
+ * <p>This is useful for providing a way for users to use sort features
+ * that are not otherwise implemented in Hibernate Search.
+ *
+ * @author Yoann Rodiere
+ */
+public class ArbitrarySort extends Sort {
+
+	private String fieldName;
+
+	private Object sortDescription;
+
+	public ArbitrarySort(String fieldName, String sortDescription) {
+		super( fieldName );
```

Comment:
I like the simplicity of approach - thinking of end users - of having both parameters.

The fact that the first one might not be a real "field" makes me think we might want to rename the parameter, more than remove it?

This deserves further discussion but I think it's time to merge this, we can amend it shortly as needed.